mart/attack/adversary.py

```diff
@@ -19,29 +25,71 @@ class AdversaryCallbackHookMixin(Callback):
 
     callbacks = {}
 
-    def on_run_start(self, adversary, input, target, model, **kwargs):
+    def on_run_start(
+        self,
+        adversary: Callback,
+        input: Union[torch.Tensor, tuple],
+        target: Union[torch.Tensor, Dict[str, Any], tuple],
+        model: SequentialDict,
```

Comment:
I think this should be more generally typed as a `torch.nn.Module`.

Reply:
At least, I don't think we make assumptions about `model` being a `SequentialDict`.